src/core/message.ts

```diff
@@ -417,7 +417,13 @@ export class DefaultMessage implements Message {
     return this.equal(message);
   }
 
-  with(event: ChatMessageEvent | MessageReactionSummaryEvent): Message {
+  with(event: Message | ChatMessageEvent | MessageReactionSummaryEvent): Message {
+    // If event doesn't have the property "type", then it's just a message
+    if (!('type' in event)) {
```

Comment:
Is not having `type` the best param here? Events are typically `{ type, message }` similar so it works, but if we ever have a message `type` (I hope not!) this gets weird.

How about checking if there's a `serial` since messages must have `serial` end events shouldn't (it'll be nested)?

Reply:
TypeScript actually catches this - because it knows that `type` isn't in `Message`. But `serial` works just as well.